### PR TITLE
Inital module commit

### PR DESCRIPTION
**Pull request by bartoszkubicki**

### CODE REVIEW

File: Api/RootCategoryProviderInterface.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Class RootCategoryProviderInterface
+ * @package LizardMedia\AllProductsListing\Api
+ */
+interface RootCategoryProviderInterface
+{
+    /**
+     * @return CategoryInterface
+     * @throws NoSuchEntityException
+     */
+    public function provide() : CategoryInterface;
```
Comment (maciejslawik) [inline Api/RootCategoryProviderInterface.php]: I would change the name of the method to something more descriptive, like 'getRootCategory'

File: Controller/Index/Index.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    public function execute()
+    {
+        try {
+            $category = $this->initRootCategory();
+            $this->layerResolver->create(Resolver::CATALOG_LAYER_CATEGORY);
+        } catch (\Exception $exception) {
+            return $this->resultForwardFactory->create()->forward('noroute');
+        }
+
+        $this->catalogSession->setLastViewedCategoryId($category->getId());
```
Comment (maciejslawik) [inline Controller/Index/Index.php:130]: You do this twice, why? (The first occurance is in the 'initRootCategory' method

File: Model/ListingPageProcessor.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+     * @var Design
+     */
+    private $catalogDesign;
+
+    /**
+     * @var CategoryUrlPathGenerator
+     */
+    private $categoryUrlPathGenerator;
+
+
```
Comment (maciejslawik) [inline Model/ListingPageProcessor.php]: Inconsistent number of new lines

File: Test/Integration/ModuleConfigTest.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * @test
+     */
+    public function testTheModuleIsConfiguredAndEnabled()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $moduleList = $objectManager->create(ModuleList::class);
+        $this->assertTrue($moduleList->has(self::MODULE_NAME));
+    }
+}
```
Comment (maciejslawik) [inline Test/Integration/ModuleConfigTest.php]: No new line at the end of file

File: Controller/Index/Index.php (hunk 2 above)
Comment (bartoszkubicki) [reply Controller/Index/Index.php:130]: setLastVisitedCategoryId vs setLastViewedCategoryId

File: Test/Integration/ModuleConfigTest.php (hunk 4 above)
Comment (bartoszkubicki) [reply Test/Integration/ModuleConfigTest.php]: corrected

**Pull request merged**